src/main/java/com/squareup/kotlinpoet/TypeSpec.kt

```diff
@@ -134,7 +136,7 @@ class TypeSpec private constructor(builder: TypeSpec.Builder) {
         val types = listOf(superclass).filter { it != ANY }.map {
           CodeBlock.of("%T(%L)", it, superclassConstructorParameters.joinToCode())
         }
-        val superTypes = types + superinterfaces.map { CodeBlock.of("%T", it) }
+        val superTypes = types + superinterfaces.map { CodeBlock.of("%T", it) } + superinterfaceDelegates.map { CodeBlock.of("${it.type} by ${it.name}") }
```

Comment:
Type of the superinterface and type of the parameter aren't necessarily the same, take this case for example:

```kotlin
class A : Consumer<Byte> {
  override fun accept(t: Byte) {
  }
}

class B(val a : A): Consumer<Byte> by a
```